Non-member arsenal limits management dialog

jaj22 opened the pull request:
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [X] Enhancement

### What have you changed and why?
This PR gives the commander an interface to manually adjust the non-member limit for each item in the arsenal. This isn't essential with no-unlocks (you could use flat or category-dependent limits instead), but it enables some management flexibility. Some uses:

- Items that are uncommon but also not popular could have their limits reduced towards zero, to give non-members a greater equipment choice, and something non-terrible to use if the popular items run out.
- Some common and effective items could have higher limits set to increase the chance that the AIs use them.
- Items that you simply don't want non-members using for whatever reason could have their limits raised.

This is still experimental. It'll likely need reworking to fit in with the new Antistasi GUI system. Currently it hooks onto the recently-blank button under the commander menu. Other stuff that probably needs adding:

- Limit data should be saved & loaded.
- Default limits should be parameter based.
- Reset button for limits (in a category? all?).
- Ideally remove the arsenal type index magic numbers, but that might not be reasonable due to how config is loaded.    

### Please specify which Issue this PR Resolves.
Related to the grand no-unlocks project. Should probably write something for that.

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

See above.

### How can the changes be tested?
Y -> Commander -> arsenal limits. Play around.

Bob-Murphy commented on A3A/addons/core/functions/Dialogs/ArsenalLimits/arsenalLimitsDialogs.hpp:
```diff
@@ -0,0 +1,257 @@
+                    style = ST_PICTURE + ST_KEEP_ASPECT_RATIO;
+                    idc = IDC_typesBase + 0;
+                    text="\A3\Ui_f\data\GUI\Rsc\RscDisplayArsenal\PrimaryWeapon_ca.paa";
+                    tooltip="$STR_A3_RscDisplayArsenal_tab_PrimaryWeapon";
```
The tooltip strings, do they exist?
Can't find them in unstable and don't see stringtable changes.

HakonRydland reviewed (changes requested):
a bunch of localization that needs to be done, and a few "requests" for approach changes, only real need to change is the data parsing from text, and we should be using A3A GUI resources, other than that the code looks solid.

Admittedly i haven't actually tested in arma so I'm willing to accept not switching to the new style if its greatly not desired.

jaj22 commented on A3A/addons/core/functions/Dialogs/ArsenalLimits/arsenalLimitsDialogs.hpp:
```diff
@@ -0,0 +1,257 @@
+            idcLeft = IDC_arrowMinus;
+            idcRIght = IDC_arrowPlus;
+            drawSideArrows=1;
+            disableOverflow=1;
```
Yeah, stops text overflowing into the next box along. Potentially important here because some mods can have enormous weapon names.

jaj22 commented on A3A/addons/core/functions/Dialogs/ArsenalLimits/fn_arsenalLimits_listButton.sqf:
```diff
@@ -0,0 +1,14 @@
+
+private _listBox = _display displayCtrl IDC_listBox;
+private _stepButton = _display displayCtrl IDC_stepButton;
+private _stepCount = [1, 5] select ("5" in ctrlText _stepButton);
```
It's just a button, not an LNB entry, so there's no value/data. I eventually figured that you can setVariable on the display to store this sort of data, although it does make the code so much longer that I'll need an extra function just for the stupid toggle button.

jaj22 commented:
Ok, refactored it into the new-GUI codebase. Stuff done:
- Refactored IDD/IDCs, code and file arrangements into the new GUI coding style.
- Stringtable entries for the arsenal limits dialog itself.
- Removed all string parsing, at great cost to sanity because onLoad + call is fucked.

Stuff not done:
- Stringtable entries for the old Y dialog, because they'll be gone soon anyway.
- Any attempt at a style merge with the other new UI stuff, because life is short. I'll think about it again once the parent functionality is there.

HakonRydland commented on A3A/addons/GUI/dialogues/arsenalLimitsDialog.hpp:
```diff
@@ -0,0 +1,258 @@
+
+#include "ids.inc"
+
+import RscText;
+import RscButton;
+import RscListNBox;
+import RscControlsGroup;
```
we have our own base resources, see `controls.hpp` included **before** `dialogs.hpp` in `config.cpp`

HakonRydland reviewed (approved):
looks mostly good, ***should*** be using the A3A resources but i will let it go for now as it is not tied to functionality but rather uniformity of UI
